relay-server/src/actors/global_config.rs

```diff
@@ -0,0 +1,60 @@
+use std::sync::Arc;
+
+use relay_dynamic_config::GlobalConfig;
+use relay_system::{Addr, Interface, Service};
+use tokio::sync::mpsc;
+
+use crate::actors::project_cache::ProjectCache;
+
+// No messages are accepted for now.
+#[derive(Debug)]
+pub enum GlobalConfigMessage {}
+
+impl Interface for GlobalConfigMessage {}
+
+/// Helper type to forward global config updates.
+pub struct UpdateGlobalConfig {
+    global_config: Arc<GlobalConfig>,
+}
+
+/// Service implementing the [`GlobalConfig`] interface.
+///
+/// The service is responsible to fetch the global config appropriately and
+/// forward it to the services that require it.
+#[derive(Debug)]
+pub struct GlobalConfigService {
+    project_cache: Addr<ProjectCache>,
+}
+
+impl GlobalConfigService {
+    pub fn new(project_cache: Addr<ProjectCache>) -> Self {
+        GlobalConfigService { project_cache }
+    }
+
+    /// Forwards the given global config to the services that require it.
+    fn update_global_config(&mut self, new_config: UpdateGlobalConfig) {
+        self.project_cache.send(new_config.global_config);
+    }
+}
+
+impl Service for GlobalConfigService {
+    type Interface = GlobalConfigMessage;
+
+    fn spawn_handler(mut self, _rx: relay_system::Receiver<Self::Interface>) {
+        tokio::spawn(async move {
+            relay_log::info!("global config started");
+
+            let (_config_tx, mut config_rx) = mpsc::unbounded_channel();
+            loop {
+                tokio::select! {
+                    biased;
+
+                    Some(message) = config_rx.recv() => self.update_global_config(message),
+                    // TODO(iker): request global config
+                    else => break,
+                }
+            }
```

Comment:
What other service will hold `_config_tx`? Wouldn't this be a message that we define in `UpdateGlobalConfig`?

On a higher level, what's the purpose of a service that only forwards a message? Couldn't whoever sends the update request send it straight to the `ProjectCache`?

Or will this service fetch the project config in a cron-job-like manner, and then call `update_global_config` itself?

TL;DR I don't really understand what's going on in this PR. Could you provide more information on the design, or add a bit more functionality to the PR so that it explains the design itself?

Reply:
Added a small clarification in https://github.com/getsentry/relay/pull/2300/files#r1262827148, let me know if it's still unclear. My idea was to separate the work into fetching and forwarding the global config, and this PR does the forwarding.